Improve documentation

ruuda opened the pull request:
I failed to get this plugin working as described in DOCS.md, so I updated it to reflect what I had to do to make it work.

This builds upon #3.

ruuda committed:
improve documentation

Currently this plugin does not work without setting the 'image' key,
and the correct image for the secret is 'plugins/sftp-cache', not
'sftp-cache'.

tboerger commented:
My pr have been merged, so please rebase your pr

tboerger commented:
https://github.com/drone-plugins/drone-sftp-cache/pull/6 will handle the directory creation

ruuda committed:
do not suggest to create a cache dir for the repo

Soon the plugin should create the directory if it does not exist, there
is no need to create it manually.

ruuda commented:
> My pr have been merged, so please rebase your pr

Done.

> IMHO we should skip that, those values are most of the time part of the yaml and you are not forced to use secrets for that.

I agree, but currently using the yaml is broken.

> I would prefer a pull request that creates the directories.

That would be convenient if it succeeds, but it will not be possible in general, because the configured user might not have permissions to do so. For example, the canonical location `/var/cache` is owned by root on my system. I think it is still worth pointing this out. I did remove the final `mkdir`, as creating that one automatically should succeed.

tboerger commented:
For all the plugins we provide the secrets command only for critical values like passwords. Why should the yaml be broken?

tboerger commented:
And for the permissions, just take a directory that is really writable?

ruuda committed:
do not put config unnecessarily in secrets in docs

ruuda commented:
> For all the plugins we provide the secrets command only for critical values like passwords. Why should the yaml be broken?

When I tried it, putting the server and path in the yaml didn’t work, but now that I tried to verify, it does. I must have been doing something else wrong. I updated the pull request.

> And for the permissions, just take a directory that is really writable?

That would pretty much mean putting it in `/home/username/somewhere`. I don’t think the plugin should dictate how users should organise their file system. Either way, it would be good to point out that the directories can only be created automatically if the user has the right permissions.

tboerger commented on DOCS.md:
```diff
@@ -46,14 +46,28 @@ drone sign octocat/hello-world
 See [secrets](http://readme.drone.io/0.5/usage/secrets/) for additional
 information on secrets
 
+On the remote server, ensure that the cache directory exists:
```
Can you move that to the top with the headline `## Preparation`?

tboerger commented:
Sorry, I forgot your PR... We have at least updated the `plugins/*` image references.

ruuda commented:
Looks like this is no longer necessay then.
